game-engine/src/main/scala/za/co/entelect/challenge/game/contracts/game/CarGamePlayer.scala

```diff
@@ -58,20 +73,49 @@ class CarGamePlayer(health: Int, var score: Int, gamePlayerId: Int, var speed: I
     }
 
     def hasTurnedThisRound(): Boolean = {
-        return state == Config.TURNING_LEFT_PLAYER_STATE || state == Config.TURNING_RIGHT_PLAYER_STATE
+        return statesThatOccurredThisRound.contains(Config.TURNING_LEFT_PLAYER_STATE) || statesThatOccurredThisRound.contains(Config.TURNING_RIGHT_PLAYER_STATE)
     }
 
-    def hitItem(item: Int): Unit = {
+    def hitItem(item: Int, playerIsInert: Boolean): Unit = {
+        if(playerIsInert) return
         if (isLizarding) return
         if (item == Config.MUD_MAP_OBJECT) {
+            damage += Config.DAMAGE_MUD;
+            reduceMaxAllowableSpeed()
             hitMud()
         }
         if (item == Config.OIL_SPILL_MAP_OBJECT) {
+            damage += Config.DAMAGE_OIL;
```

Comment:
Why not make the damage mutation part of the `hitX` methods?

Reply:
Not 100% sure I follow @KyleMcAllister .
If you make damage modifier part of the `hit*` method, then:
- for `hitMud`, you *know* you are hitting mud, thus you know which config value to select. Same for all other damage inflicting items
- making it part of the relevant `hit*` method ensures that terrain affects are always applied (well, as long as the relevant `hit*` method is called).

I'm thinking,for example of the cyberTruck-boost bug, where the relevant method was not called, and speed was directly modified, leading to the situation where boost was not canceled after cyber hit, causing player speed to (potentially) jump back to 9 when boost counter reached 0, irrespective of current player speed.

To be clear, I was not suggesting create one "handle all hits" method - I'm just sayind merge the damage modifier for oil into the `hitOil` method, damage for mud into `hitMud` etc.